### PR TITLE
Category selection

### PR DESCRIPTION
**Pull request by zoef-yt**
[CHECK ME ONLY](https://github.com/zoef-yt/game-plex-react/pull/7/commits/dce4c6e1c5819574892e2e756e609bd12dcbd176)
[Preview link](https://deploy-preview-7--game-plex-zoef.netlify.app/videos)
In this pr, I added

- category selection feature
- You can select any category from the home page and get the selected category video on the videos pages.
- You can change or select a category on the videos page too.

### CODE REVIEW

**Review by sanu29 — APPROVED** [review]
All good. Clean readable code. Good to merge

File: src/Context/HistoryContext/HistoryContext.jsx [hunk 1]
```diff
@@ -4,27 +4,62 @@ import { useAxios } from '../../CustomHooks/useAxios';
...
-	]);
-	return <HistoryContext.Provider value={{ historyList }}>{children}</HistoryContext.Provider>;
+	const [historyList, useHistoryList] = useState([]);
+
+	const { response: historyResponse, operation: historyServerCall, error: historyError } = useAxios();
+
+	const token = localStorage.getItem('token');
+
+	useEffect(() => {
+		if (historyResponse != null && historyResponse.history) {
```
Comment (itsmebipul00) [inline src/Context/HistoryContext/HistoryContext.jsx:14]: I think you could do this with optional chaining and wrap it with double not or Boolean

**Review by itsmebipul00 — APPROVED** [review]
Nice work Zoef ! Good to merge

File: src/Context/FilterContext/FilterContext.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { useContext, createContext, useState, useEffect, useReducer } from 'react';
+
+const initialData = {
+	selectedCategory: '',
+	dataToShow: '',
```
Comment (muskanbagrecha) [inline src/Context/FilterContext/FilterContext.jsx:5]: dataToShow is going to be an array right? Then why initialize with ""?

File: src/Pages/VideoListingPage/components/VideoListingSection.jsx [hunk 3]
```diff
@@ -1,12 +1,36 @@
-import { useVideo } from '../../../Context';
+import { useEffect } from 'react';
+import { useFilter, useVideo } from '../../../Context';
 import { VideoCard } from './VideoCard';
 const VideoListingSection = () => {
 	const { videos, loadingVideos } = useVideo();
+	const { filterState, filterDispatch } = useFilter();
+	useEffect(() => {
```
Comment (muskanbagrecha) [inline src/Pages/VideoListingPage/components/VideoListingSection.jsx:7]: Will this not work if dep is []?

File: src/Pages/VideoListingPage/components/VideoListingSection.jsx [hunk 4]
```diff
@@ -1,12 +1,36 @@
...
+			});
+	}, [videos]);
+	const isDataToShowEmpty = filterState?.dataToShow;
+
+	const getFilteredData = (state) => {
+		const selectedCategoryVideos = selectedCategory(state.selectedCategory, state.dataToShow);
+		return selectedCategoryVideos;
+	};
+
+	const selectedCategory = (selectedCategory, dataToShow) => {
```
Comment (muskanbagrecha) [inline src/Pages/VideoListingPage/components/VideoListingSection.jsx:21]: Please improve the function naming. Also I dont recommend using the same name for args and function.

File: src/Pages/VideoListingPage/components/VideoListingSection.jsx [hunk 5]
```diff
@@ -1,12 +1,36 @@
...
+		return selectedCategory === '' || selectedCategory === 'ALL'
+			? dataToShow
+			: [...dataToShow].filter((video) => video.categoryName === selectedCategory);
+	};
+
 	return (
 		<>
-			<div className='video-listing-section text-grey'>Showing result:- {videos.length}</div>
+			<div className='video-listing-section text-grey'>Showing result:- {isDataToShowEmpty && getFilteredData(filterState).length}</div>
```
Comment (muskanbagrecha) [inline src/Pages/VideoListingPage/components/VideoListingSection.jsx:29]: I am kind of confused about this line. If data is empty, fetch it and show the length. If not empty?

**Review by muskanbagrecha — APPROVED** [review]
Code looks great. It was a small PR, did not find much issues. You can look into the comments.

File: src/Pages/VideoListingPage/components/VideoListingSection.jsx (hunk 5 above)
Comment (zoef-yt) [reply src/Pages/VideoListingPage/components/VideoListingSection.jsx:29]: Yeah here it should be 0. Thanks

File: src/Context/FilterContext/FilterContext.jsx (hunk 2 above)
Comment (zoef-yt) [reply src/Context/FilterContext/FilterContext.jsx:5]: Done

File: src/Pages/VideoListingPage/components/VideoListingSection.jsx (hunk 4 above)
Comment (zoef-yt) [reply src/Pages/VideoListingPage/components/VideoListingSection.jsx:21]: Done

**Pull request merged**